spec/api-session-spec.ts

```diff
@@ -31,6 +31,30 @@ describe('session module', () => {
     });
   });
 
+  describe('session.fromPartition(partition, options)', () => {
```

Comment:
We also need test to make sure default values are used when quotasize option is not defined.
We also need negative tests. Like defining negative numbers, too large numbers, 0 etc.

Reply:
I did try to add a test which queries the disk free space of the specified path. However, this test then cannot be run on platforms and hence would fail. The procedure I followed was to use the childProcess to run `df <path>`, but that has to be further extracted and again this would not work on Windows which then would then fail the test and not allow the patch to land upstream.